### PR TITLE
Update sewing-kit

### LINKED ISSUE #1701: [v4] Check for sewing-kit update before release v4
See #1695

### PR DESCRIPTION
**Pull request by BPScott**
This will now warn on unsafe lifecycle hooks like componentWillReceiveProps

Fixes #1701.

### CODE REVIEW

File: src/components/Card/tests/Card.test.tsx [hunk 1]
```diff
@@ -6,8 +6,10 @@ import {contentContextTypes} from '../../../types';
 
 describe('<Card />', () => {
   it('has a child with contentContext', () => {
-    const Child: React.SFC<{}> = (_props, context) =>
-      context.withinContentContainer ? <div /> : null;
+    const Child: React.SFC = (_props, context) => {
```
Comment (BPScott) [inline src/components/Card/tests/Card.test.tsx]: ping @cartogram This sort of thing feels like a false positive on `shopify/jest/no-if`. would you be able to look into it?

Comment (cartogram) [reply src/components/Card/tests/Card.test.tsx]: Yeah, it is. Can you copy and paste this example into an issue on the eslint-plugin-shopify?

For now, keep the inline disable and when I fix the rule, the disable comment should be flagged but the no-disable-comments rule and you can remove it then. Thanks Ben!

Comment (BPScott) [reply src/components/Card/tests/Card.test.tsx]: Done! https://github.com/Shopify/eslint-plugin-shopify/issues/311

File: src/components/TextField/TextField.tsx [hunk 2]
```diff
@@ -143,6 +143,7 @@ class TextField extends React.PureComponent<CombinedProps, State> {
   constructor(props: CombinedProps) {
     super(props);
 
+    // eslint-disable-next-line react/state-in-constructor
```
Comment (AndrewMusgrave) [inline src/components/TextField/TextField.tsx]: We could pull this one out of the constructor if we wanted

**Pull request merged**